src/aggregated_range_proof/dealer.rs

```diff
@@ -0,0 +1,174 @@
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use curve25519_dalek::traits::Identity;
+use generators::GeneratorsView;
+use inner_product_proof;
+use proof_transcript::ProofTranscript;
+use std::clone::Clone;
+use util;
+
+use super::messages::*;
+
+/// Dealer is an entry-point API for setting up a dealer
+pub struct Dealer {}
+
+impl Dealer {
+    /// Creates a new dealer with the given parties and a number of bits
+    pub fn new(
+        transcript: &mut ProofTranscript,
+        n: usize,
+        m: usize,
+    ) -> Result<DealerAwaitingValues, &'static str> {
```

Comment:
Should this return a `Result` if it's always going to be `Ok`?

Reply:
I was setting up the function to potentially return an error - but then I moved the error handling to PR #57 so this isn't needed here.